src/test/kotlin/org/rust/ide/intentions/IfLetToMatchIntentionTest.kt

```diff
@@ -636,4 +636,47 @@ class IfLetToMatchIntentionTest : RsIntentionTestBase(IfLetToMatchIntention()) {
             }
         }
     """)
+
+    fun `test irrefutable pattern`() = doAvailableTest("""
+        struct Id(u32);
+        struct S { a: Id, b: u32 }
+        fn foo(s: S) {
+            if let S { a: Id(ref name), .. } = s/*caret*/ {
+                let _x = a;
+            }
+        }
+    """, """
+        struct Id(u32);
+        struct S { a: Id, b: u32 }
+        fn foo(s: S) {
+            match s {
+                S { a: Id(ref name), .. } => {
+                    let _x = a;
+                }
+            }
+        }
+    """)
+
+    fun `test irrefutable pattern with else`() = doAvailableTest("""
```

Comment:
Hmmm, I think we should keep else-branch here. @Undin what do you think?

Reply:
I agree that we should keep else branch if a user already has it

Reply:
BTW, it would be great to have an inspection to annotate such redundant `else`